modules/fee/handler.go

```diff
@@ -45,40 +46,72 @@ func (SimpleFeeMiddleware) Name() string {
 }
 
 // CheckTx - check the transaction
-func (h SimpleFeeMiddleware) CheckTx(ctx basecoin.Context, store state.SimpleDB, tx basecoin.Tx, next basecoin.Checker) (res basecoin.Result, err error) {
-	return h.doTx(ctx, store, tx, next.CheckTx)
+func (h SimpleFeeMiddleware) CheckTx(ctx basecoin.Context, store state.SimpleDB, tx basecoin.Tx, next basecoin.Checker) (res basecoin.CheckResult, err error) {
+	fee, err := h.verifyFee(ctx, tx)
+	if IsSkipFeesErr(err) {
```

Comment:
Personal style but I prefer to convey that we've encountered an error, it'd be nice to make it:
```go
if err != nil {
   if IsSkipFeesErr(err) {
      return next.CheckTx(ctx, store, tx)
   }
   return res, err
}
```